script/build.sc

```diff
@@ -9,12 +9,29 @@ import mill.scalalib.scalafmt._
 object v {
   val scala3 = "3.3.3"
   val mainargs = ivy"com.lihaoyi::mainargs:0.5.0"
-  val oslib = ivy"com.lihaoyi::os-lib:0.9.1"
+  val oslib = ivy"com.lihaoyi::os-lib:0.10.0"
   val upickle = ivy"com.lihaoyi::upickle:3.1.3"
 }
 
 object script extends RootModule with ScalaModule {
   def scalaVersion = v.scala3
   def scalacOptions = Seq("-new-syntax")
   override def ivyDeps = Agg(v.mainargs, v.oslib, v.upickle)
+
+  def nativeImage = T {
+    val jar = super.assembly()
+    val nativeImageArgs = Seq(
+      // don't fallback to using JRE, so that we can get a minimal size binary.
+      "--no-fallback"
+    )
+    val outputArgs = Seq(
+      "-jar",
+      jar.path.toString(),
+      (T.dest / "out.elf").toString()
+    )
+    os.proc(
+      Seq("native-image") ++ nativeImageArgs ++ outputArgs
+    ).call(stdout = os.Inherit, stderr = os.Inherit)
+    PathRef(T.dest / "out.elf")
+  }
```

Comment:
Move these to Nix, mill should only release assembly `jar`, while using graalvm to build them to elf.

This shared scripts can be used for migrating our other jars to native for speed up the elaboration.